src/Optimize/OptimizeBase.h

```diff
@@ -47,6 +47,8 @@ class CostFunctionBase
 
   virtual Return_t Params(int i) const = 0;
 
+  virtual std::string getName(int i) const = 0;
```

Comment:
Try to be verbose, "getParamName"? can the return type "const std::string&"?

Reply:
I changed the name of the function to getParamName.   I don't think returning a std::string reference will be any better than the value, due to Return Value Optimization.

Reply:
Even with RVO, there is one copy made while consistently using "const std::string&" since the initial return should ensure no copy. I will stop insist on this since this part is not performance critical.